src/WireMock.Net/Http/HttpRequestMessageHelper.cs

```diff
@@ -33,12 +33,17 @@ internal static HttpRequestMessage Create(IRequestMessage requestMessage, string
             MediaTypeHeaderValue.TryParse(value, out contentType);
         }
 
-        httpRequestMessage.Content = requestMessage.BodyData?.DetectedBodyType switch
+        var bodyData = requestMessage.BodyData;
+        var bodyType = bodyData?.DetectedBodyTypeFromContentType?.ToNullable()
```

Comment:
Probably better and more readable to add a method `BodyType GetBodyType()` to the BodyData class.
``` c#
BodyType GetBodyType()
{
    if (DetectedBodyTypeFromContentType != null)
    {
        return DetectedBodyTypeFromContentType;
    }

    // and so on.....
}
```

Reply:
Problem is, here, we get `IBodyData` from the `requestMessage`. But I can write a class extension for that interface :)

Reply:
Ah. Just FYI, we can't simply check `null` as the value of `DetectedBodyTypeFromContentType` is not null, but `BodyType.None`. That's why I had to write an extension method for assuming `null` for this value.